File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/ListView/Features/ListViewControlPanel.vb
Partial Public Class ListViewControlPanel
    Inherits UserControl
    Private mstrPrevFrom As String = ""
    Private mstrPrevSubject As String = ""
    Private mblnPrevImportant As Boolean = False
    Private mblnPrevRead As Boolean = False
    Private mblnPrevAttachemnts As Boolean = False
    Private mstrPrevMailSize As String = ""

    Public Sub New()
        InitializeComponent()
    End Sub

    Friend ReadOnly Property IsPanelDirty() As Boolean
        Get
            Return mblnPrevAttachemnts <> Attachemnts OrElse mstrPrevSubject <> Subject OrElse mstrPrevFrom <> From OrElse mstrPrevMailSize <> MailSize OrElse mblnPrevImportant <> Important OrElse mblnPrevRead <> Read
        End Get
    End Property

    Friend Sub SetPrevValues()
        mblnPrevAttachemnts = Attachemnts
        mstrPrevSubject = Subject
        mstrPrevFrom = From
        mstrPrevMailSize = MailSize
        mblnPrevImportant = Important
        mblnPrevRead = Read
    End Sub

    Friend Property Subject() As String
        Get
            Return textBoxSubject.Text
        End Get
        Set(ByVal value As String)
            textBoxSubject.Text = value
        End Set
    End Property

    Friend Property From() As String
        Get
            Return textBoxFrom.Text
        End Get
        Set(ByVal value As String)
            textBoxFrom.Text = value
        End Set
    End Property

    Friend Property MailSize() As String
        Get
            Return textBoxSize.Text
        End Get
        Set(ByVal value As String)
            textBoxSize.Text = value
        End Set
    End Property

    Friend Property Attachemnts() As Boolean
        Get
            Return checkBoxAttachments.Checked
        End Get
        Set(ByVal value As Boolean)
            checkBoxAttachments.Checked = value
        End Set
    End Property

    Friend Property Important() As Boolean
        Get
            Return checkBoxImportant.Checked
        End Get
        Set(ByVal value As Boolean)
            checkBoxImportant.Checked = value
        End Set
    End Property

    Friend Property Read() As Boolean
        Get
            Return checkBoxRead.Checked
        End Get
        Set(ByVal value As Boolean)
            checkBoxRead.Checked = value
        End Set
    End Property
End Class

File: Sources 4.5.2/Gizmox.WebGUI.Forms.Catalog/Categories/DataControls/ListViewControlPanel.cs
#region Using

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;

using Gizmox.WebGUI.Common;
using Gizmox.WebGUI.Forms;

#endregion

namespace Gizmox.WebGUI.Forms.Catalog.Categories.DataControls
{
    [Serializable]
    public partial class ListViewControlPanel : UserControl
    {
        private string mstrPrevFrom = "";
        private string mstrPrevSubject = "";
        private bool mblnPrevImportant = false;
        private bool mblnPrevRead = false;
        private bool mblnPrevAttachemnts = false;
        private string mstrPrevMailSize = "";

        public ListViewControlPanel()
        {
            InitializeComponent();
        }

        internal bool IsPanelDirty
        {
            get
            {
                return
                    mblnPrevAttachemnts != Attachemnts ||
                    mstrPrevSubject != Subject ||
                    mstrPrevFrom != From ||
                    mstrPrevMailSize != MailSize ||
                    mblnPrevImportant != Important ||
                    mblnPrevRead != Read;
            }
        }

        internal void SetPrevValues()
        {
            mblnPrevAttachemnts = Attachemnts;
            mstrPrevSubject = Subject;
            mstrPrevFrom = From;
            mstrPrevMailSize = MailSize;
            mblnPrevImportant = Important;
            mblnPrevRead = Read;
        }

        internal string Subject
        {
            get
            {
                return textBoxSubject.Text;
            }
            set
            {
                textBoxSubject.Text = value;
            }
        }

        internal string From
        {
            get
            {
                return textBoxFrom.Text;
            }
            set
            {
                textBoxFrom.Text = value;
            }
        }

        internal string MailSize
        {
            get
            {
                return textBoxSize.Text;
            }
            set
            {
                textBoxSize.Text = value;
            }
        }

        internal bool Attachemnts
        {
            get
            {
                return checkBoxAttachments.Checked;
            }
            set
            {
                checkBoxAttachments.Checked = value;
            }
        }

        internal bool Important
        {
            get
            {
                return checkBoxImportant.Checked;
            }
            set
            {
                checkBoxImportant.Checked = value;
            }
        }

        internal bool Read
        {
            get
            {
                return checkBoxRead.Checked;
            }
            set
            {
                checkBoxRead.Checked = value;
            }
        }
    }
}